SimplifiedTests/MockServiceUsingAxisItemDownloader.swift

```diff
@@ -0,0 +1,31 @@
+//
+//  MockServiceUsingAxisItemDownloader.swift
+//  OpenEbooksTests
+//
+//  Created by Raman Singh on 2021-05-17.
+//  Copyright © 2021 NYPL Labs. All rights reserved.
+//
+
+import Foundation
+@testable import SimplyE
+
+class MockServiceUsingAxisItemDownloader {
```

Comment:
I think the name of this class is kinda confusing.
Can it be named something like `NYPLAxisItemDownloaderMockResponseService / NYPLAxisItemDownloaderMockSucceedService`? Or anything describe this class better.
@ettore any thoughts?

Reply:
I'm not sure... I don't think this is mocking a service (it's not implementing any protocols), it just provides functionality to other mocks as part of their own class hierarchy. Mock ideally should be as simple as possible to avoid getting into situations where the flow of the test is hard to follow.

Maybe I would call it NYPLDownloadRunnerMock since it's triggering the downloader? but I'm actually not sure how this works to make a good suggestion.

Reply:
NYPLDownloadRunnerMock makes more sense. This class acts as a base class for NYPLAxisMetadataServiceMock,  NYPLAxisPackageServiceMock, & NYPLAxisLicenseServiceMock. If you believe this complicates the code I'll be more than happy to make changes to it.

Reply:
generally speaking in tests is totally fine to repeat the same code over and over. The objective is to write tests that are easy to understand -- we don't want to look for logic bugs in tests, essentially. I do get it that like in this case there is some test logic that is repeated.

But no I'm not asking you to change it because I don't have better suggestions right now (I would have to look at it more deeply and I can't right now).

For the future though I would limit code reuse to things like helper functions but not much else. I'd try to avoid any logic like conditional branches, loops, subclassing-- not always possible but something to strive for.